File: services/ProductService/Models/Category.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProductService.Models
{
    [Table("TbCategories")]
    public class Category
    {
        [Key]
        public int CategoryId { get; set; }
        [Required]
        public string CategoryName { get; set; } = string.Empty;
        public bool IsActive { get; set; } = true;
        public int LastUpdatedBy { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime LastUpdatedOn { get; set; }
    }
}

File: frontend/src/app/models/Category.ts
// Category Model: Structure of a category as per the database
export interface Category {
  categoryId: number;
  categoryName: string;
  isActive: boolean;
  createdBy: number;
  lastUpdatedBy: number;
  createdOn: string; // ISO date string
  lastUpdatedOn: string; // ISO date string
}
